### DISCUSSION TITLE
Twind doesn't work well with PrismJs

### DISCUSSION THREAD

**Discussion by srid** [tw-in-js/twind#180] [Q&A] [answered]
It looks like Twind shim is resetting the syntax highlighting stuff injected by https://prismjs.com/ and I see these warnings in the console:

![image](https://user-images.githubusercontent.com/3998/116607857-8320ef80-a900-11eb-807b-2ecfaddd81c0.png)

You can see this in action here: https://ema.srid.ca/start/tutorial

Note that code blocks are syntax highlighted on page load, and then it is undone after a fraction of a second (when twind resets them).

Is there a way avoid this?

I should clarify that this is a statically generated site, so no JS is otherwise involved.

**Comment by sastan** [comment]
I tried your page without JS (no shim) and I do not see the syntax highlight. The classes are not removed from the elements. Are you sure you are sending the CSS for the highlight?

**Reply by sastan** [reply]
You can get rid of the warnings with: `setup({mode: 'silent'})`

**Reply by srid** [reply]
Well, the static site only injects a pre/code block with "language-haskell" class. https://prismjs.com/ is supposed to take over from that point to actually do the syntax highlighting.

The strange thing is that this issue only exists in the tutorial page. On other pages, it works: for example - https://ema.srid.ca/guide/routes is syntax highlighted (though the warnings -- which appear to come from twind complaining of prismjs's attrs -- remain).

**Reply by srid** [reply]
I feel like this is tricky to solve using the JS shim approach, so I'm considering doing a compilation on the CLI (during static site generation). I don't think twind supports it, but windicss CLI does.

Compiling CSS also makes the site render correctly with `<noscript>`.

**Reply by sastan** [reply]
There is [@twind/cli](https://github.com/tw-in-js/twind-cli) in case you want to try that one.

**Comment by lukejacksonn** [comment]
Hey I don't know too much about prismjs but I just inspected the source of the site you posted and it seems like the content of the code block is just a string on the Tutorial page whereas on other pages the string is parsed and compiled into the appropriate markup for syntax highlighting.

Tutorial page:

![image](https://user-images.githubusercontent.com/1457604/117728725-b6f1f400-b1e1-11eb-980e-8413b8f3a549.png)

Other Page:

![image](https://user-images.githubusercontent.com/1457604/117728927-033d3400-b1e2-11eb-9ed3-b4ec1a5d57d5.png)

Syntax highlighting seems to work on both the initial and subsequent loads of all other pages other than the Tutorial. Assuming you are using Twind on all pages, my conclusion is that this issue is with that specific page and/or SSR rather than with Twind.

Did you ever find what was up?

**Reply by srid** [reply]
@lukejacksonn That's just Chrome inspector's visualization. If you "view source" the raw HTML, it is generated correctly (i.e., no wrapping quotes). The HTML is supposed to contain plain text in `<code>` blocks, and then prismJS is suppose to traverse the DOM and inject all those `token`-classed `<span>`'s, as you see in your second screenshot. The problem is, in the tutorial page, something weird happens - presumably because (judging from the "UNKNOWN DIRECTIVE" warnings) twind is traversing the prismJS generated DOM and *deleting* all those prismJS-generates span's.

![image](https://user-images.githubusercontent.com/3998/117745446-7cd02480-b1d8-11eb-894e-bccb67865e2b.png)

**Reply by srid** [reply]
Turns out this is *not* Twind's fault (at least on this page). PrismJS fails highlighting the Haskell code blocks if there is a `bash` code block the same page for some reason.

Result after fixing: https://ema.srid.ca/start/tutorial

Commit fix: https://github.com/srid/ema/commit/8e5bd31dfeec30b5d005185e5d9a5749da2571be

**Comment by srid** [comment, marked as answer]
As [indicated here](https://github.com/tw-in-js/twind/discussions/180#discussioncomment-759774) this appears to be a bug in PrismJS.
